clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaConsumerServiceDelegator.java

```diff
@@ -83,39 +110,41 @@ public SharedKafkaConsumer assignConsumerFor(PubSubTopic versionTopic, PubSubTop
 
   @Override
   public void unsubscribeAll(PubSubTopic versionTopic) {
-    defaultConsumerService.unsubscribeAll(versionTopic);
-    if (consumerServiceForAAWCLeader != null) {
-      consumerServiceForAAWCLeader.unsubscribeAll(versionTopic);
-    }
+    consumerServices.forEach(kafkaConsumerService -> kafkaConsumerService.unsubscribeAll(versionTopic));
+    vtTopicPartitionPairToConsumerService.entrySet().removeIf(entry -> entry.getKey().getFirst().equals(versionTopic));
```

Comment:
Looping over a small list of consumer services is kind of okay; but here, we are iterating through all topic partitions hosted on this node to find a small set of matching topic partition, it doesn't seem like a good data structure design. Why not just topic->consumerService mapping, and let the consumerService itself take care of the subscribe or unsubscribe action?

Reply:
topic->consumerService is not correct here, as one version topic could interact with more than 1 consumer services, due to topic partition is leader or follower.

Reply:
I think @huangminchn 's comment was referring to this field: `vtTopicPartitionPairToConsumerService`.
Right now, I am not sure how important this inefficiency is as this event should be less frequent at runtime, and it will be triggered more frequently when stopping, but as different store ingestion task will trigger this function concurrently, maybe it can be fine as this function doesn't use any synchronization logic.